.github/workflows/push_docker_image.yml

```diff
@@ -0,0 +1,40 @@
+name: Publish Docker images on-demand
+on:
+  workflow_dispatch:
+    inputs:
+      image:
+        description: Image to push (either `api` or `ingestion_server`)
+        required: true
+
+jobs:
+  push:
+    name: Publish requested Docker image
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - name: Log in to GitHub Docker Registry
+        uses: docker/login-action@v1
+        with:
+          registry: https://ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Download image `${{ github.event.inputs.image }}`
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          workflow: ci_cd.yml
+          workflow_conclusion: success
+          pr: ${{github.event.pull_request.number}}
+          name: ${{ github.event.inputs.image }}
+          path: /tmp
```

Comment:
I guess the only thing to note here is Github's artifact retention policy, [which defaults to 90 days](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization). So this action could fail if run on a SHA that hasn't been built for the last 90 days.

That seems unlikely, but just in case, something to note!

Reply:
That's a great point! I hope we're not trying to run this on a 90-day old SHA 😅